File: backend/src/main/java/edu/cit/spot/dto/auth/StudentIdLoginRequest.java
package edu.cit.spot.dto.auth;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for student physical ID login requests
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class StudentIdLoginRequest {

    @NotEmpty(message = "Student physical ID is required")
    @NotBlank(message = "Student physical ID cannot be blank")
    private String studentPhysicalId;

    @NotEmpty(message = "Password is required")
    @NotBlank(message = "Password cannot be blank")
    @Size(min = 8, message = "Password must be at least 8 characters")
    private String password;
}

File: frontend_mobile/app/src/main/java/com/example/spot/model/StudentIdLoginRequest.kt
package com.example.spot.model

import com.google.gson.annotations.SerializedName

/**
 * Data class for student ID login request
 */
data class StudentIdLoginRequest(
    @SerializedName("studentPhysicalId")
    val studentPhysicalId: String,
    
    @SerializedName("password")
    val password: String
)
